render/respond.go

```diff
@@ -0,0 +1,111 @@
+package render
+
+import (
+	"bytes"
+	"encoding/json"
+	"encoding/xml"
+	"net/http"
+	"reflect"
+
+	"golang.org/x/net/context"
+)
+
+var Respond = DefaultRespond
+
+func DefaultRespond(ctx context.Context, w http.ResponseWriter, v interface{}) {
+	// Present the object.
+	if presenter, ok := ctx.Value(presenterCtxKey).(Presenter); ok {
+		v = presenter.Present(ctx, v)
+	} else {
+		v = DefaultPresenter.Present(ctx, v)
+	}
+
+	// Format data based on Content-Type.
+	switch contentType, _ := ctx.Value(contentTypeCtxKey).(ContentType); contentType {
+	case ContentTypeJSON:
+		JSON(ctx, w, v)
+	case ContentTypeXML:
+		XML(ctx, w, v)
+	default:
+		JSON(ctx, w, v)
+	}
+}
+
+func String(ctx context.Context, w http.ResponseWriter, v string) {
+	w.Header().Set("Content-Type", "text/plain; charset=utf-8")
+	status, _ := ctx.Value(statusCtxKey).(int)
+	if status == 0 {
+		status = 200
+	}
+	w.WriteHeader(status)
+	w.Write([]byte(v))
+}
+
+func HTML(ctx context.Context, w http.ResponseWriter, v string) {
+	w.Header().Set("Content-Type", "text/html; charset=utf-8")
+	status, _ := ctx.Value(statusCtxKey).(int)
+	if status == 0 {
+		status = 200
+	}
+	w.WriteHeader(status)
+	w.Write([]byte(v))
+}
+
+func JSON(ctx context.Context, w http.ResponseWriter, v interface{}) {
+	// Force to return empty JSON array [] instead of null in case of zero slice.
+	val := reflect.ValueOf(v)
+	if val.Kind() == reflect.Slice && val.IsNil() {
+		v = reflect.MakeSlice(val.Type(), 0, 0).Interface()
+	}
+
+	b, err := json.Marshal(v)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	if len(b) > 0 {
+		b = bytes.Replace(b, []byte("\\u003c"), []byte("<"), -1)
+		b = bytes.Replace(b, []byte("\\u003e"), []byte(">"), -1)
+		b = bytes.Replace(b, []byte("\\u0026"), []byte("&"), -1)
+	}
```

Comment:
interesting, there we go.. so.. I think, either call that method since this subpackage is for chi v2 which is Go 1.7-only, or copy the tip stdlib code into this function